x-pack/plugins/observability_solution/infra/public/components/asset_details/tabs/dashboards/actions/unlink_dashboard.tsx

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { EuiButtonEmpty, EuiConfirmModal } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import React, { useCallback, useState } from 'react';
+import { useKibana } from '@kbn/kibana-react-plugin/public';
+import type {
+  DashboardItemWithTitle,
+  InfraCustomDashboardAssetType,
+} from '../../../../../../common/custom_dashboards';
+import { useDeleteCustomDashboard } from '../../../hooks/use_custom_dashboards';
+import { useFetchCustomDashboards } from '../../../hooks/use_fetch_custom_dashboards';
+import { useAssetDetailsUrlState } from '../../../hooks/use_asset_details_url_state';
+
+export function UnlinkDashboard({
+  currentDashboard,
+  onRefresh,
+  assetType,
+}: {
+  currentDashboard: DashboardItemWithTitle;
+  onRefresh: () => void;
+  assetType: InfraCustomDashboardAssetType;
+}) {
+  const [isModalVisible, setIsModalVisible] = useState(false);
+  const { notifications } = useKibana();
+
+  const [, setUrlState] = useAssetDetailsUrlState();
+  const { deleteCustomDashboard, isDeleteLoading } = useDeleteCustomDashboard();
+  const { dashboards, loading } = useFetchCustomDashboards({ assetType });
+
+  const onConfirm = useCallback(
+    async function () {
+      try {
+        const linkedDashboards =
+          dashboards?.filter(
+            ({ dashboardSavedObjectId }) =>
+              dashboardSavedObjectId !== currentDashboard.dashboardSavedObjectId
+          ) || [];
```

Comment:
nit: would this be the same?

```suggestion
          (dashboards ?? []).filter(
            ({ dashboardSavedObjectId }) =>
              dashboardSavedObjectId !== currentDashboard.dashboardSavedObjectId
          );
```